### PR TITLE
Bump macOS minimum to 10.15

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Bump the minimum version of macOS supported to 10.15, which align's with Office's N-2 approach. I did this by just replying every instance of "10.14" I found in the repo. I also removed some redundant `@availability` checks.

## Changelog

[macOS] [Deprecated] - Removed support for macOS 10.14

## Test Plan

CI should be good enough. For places where I removed @availability checks, I'll do some additional testing

### CODE REVIEW

File: React/CoreModules/RCTAppearance.mm [hunk 1]
```diff
@@ -165,7 +160,7 @@ - (void)appearanceChanged:(NSNotification *)notification
 
 - (void)startObserving
 {
-  if (@available(macOS 10.14, iOS 13.0, *)) { // TODO(macOS GH#774)
+  if (@available(iOS 13.0, *)) {
     [[NSNotificationCenter defaultCenter] addObserver:self
```
Comment (Saadnajmi) [inline React/CoreModules/RCTAppearance.mm:164]: I need to verify that this still runs on macOS if I remove the macOS but of the availability check

**Review by HeyImChris — APPROVED** [review]
What about CI? The 0.64 merge was causing problems before but they should be fixed now

**Comment by Saadnajmi** [comment]
> What about CI? The 0.64 merge was causing problems before but they should be fixed now

Do you mean updating CI to use 10.15, or getting CI to pass? 

The former I'll need to double check, the latter I would count as out of scope.

**Comment by HeyImChris** [comment]
> > What about CI? The 0.64 merge was causing problems before but they should be fixed now
> 
> Do you mean updating CI to use 10.15, or getting CI to pass?
> 
> The former I'll need to double check, the latter I would count as out of scope.

The former. All the CI except the integration checks and FB's circle CI should pass. As for the integration checks, the fix is in 0.66 which is hopefully merging in soon.

File: React/CoreModules/RCTAppearance.mm (hunk 1 above)
Comment (Saadnajmi) [reply React/CoreModules/RCTAppearance.mm:164]: Still works!

**Review by amgleitman — APPROVED** [review]

**Comment by Saadnajmi** [comment]
@heyimchris Looks like the fb66 merge did a lot of the bumping, but there's still some extra `@availability` checks and the like in this PR so it's still valid

**Pull request merged**